common/common-composite-component/src/java/org/sakaiproject/component/common/edu/person/SakaiPersonManagerImpl.java

```diff
@@ -565,8 +565,10 @@ public void delete(final SakaiPerson sakaiPerson)
 		}
 		
 		
-		log.debug("getHibernateTemplate().delete(sakaiPerson);");
-		getHibernateTemplate().delete(sakaiPerson);
+		// First merge to handle potentially detached instances
+		SakaiPerson mergedPerson = getHibernateTemplate().merge(sakaiPerson);
+		log.debug("getHibernateTemplate().delete(mergedPerson);");
```

Comment:
this log message should probably be more like "Deleted SakaiPerson [{}]", sakaiPerson.getId()

Reply:
done